barretenberg/cpp/pil/vm2/bc_retrieval.pil

```diff
@@ -42,6 +44,28 @@ pol commit public_bytecode_commitment;
 // current state
 pol commit block_number;
 pol commit public_data_tree_root;
+pol commit nullifier_tree_root;
+
+// TODO: Remove this as a column when we can lookup with constants
+pol commit outer_nullifier_domain_separator;
+sel * (constants.GENERATOR_INDEX__OUTER_NULLIFIER - outer_nullifier_domain_separator) = 0;
+
+// TODO: Remove this as a column when we can lookup with constants
+pol commit deployer_protocol_contract_address;
+sel * (constants.DEPLOYER_CONTRACT_ADDRESS - deployer_protocol_contract_address) = 0;
+
+pol commit siloed_address;
+
+#[SILO_DEPLOYMENT_NULLIFIER_POSEIDON2]
+sel { outer_nullifier_domain_separator, deployer_protocol_contract_address, address, siloed_address }
+in poseidon2_hash.end { poseidon2_hash.input_0, poseidon2_hash.input_1, poseidon2_hash.input_2, poseidon2_hash.output };
```

Comment:
i was wondering if `.start` is safer for single line poseidon2 calls. I think `.start` will guarantee that the input state is 0. 

You should add the length and iv lookup here. Maybe add a TODO so we dont need another dummy column for constants.